### PR TITLE
Add write_file helper

### PR DESCRIPTION
**Pull request by RandomNoun7**
This commit adds a helper to upload a string variable as a file to a
target host.

This prevents the current requirement to use the manifest_file helper
and then a separate shell command to move the file to desired
destination path on the target host.

Closes: #323

### CODE REVIEW

**Comment by RandomNoun7** [comment]
This is a working candidate implementation. It still needs tests, but is available to gather feedback.

File: lib/puppet_litmus/puppet_helpers.rb [hunk 1]
```diff
@@ -147,6 +147,40 @@ def create_manifest_file(manifest)
...
+  # @param destination [String] The path on the target node to write the file.
+  # @return [Bool] Success. The file was succesfully writtne on the target.
+  def write_file(content, destination)
+    Honeycomb.start_span(name: 'litmus.write_file') do |span|
+      ENV['HTTP_X_HONEYCOMB_TRACE'] = span.to_trace_header
+      span.add_field('litmus.destination', destination)
+
+      require 'tmpdir'
+      target_node_name = ENV['TARGET_HOST']
+      tmp_filename = File.join(Dir.tmpdir, "litmus_tmp_#{Time.now.strftime('%Y%m%d')}_#{Process.pid}_#{rand(0x100000000).to_s(36)}")
```
Comment (DavidS) [inline lib/puppet_litmus/puppet_helpers.rb:162]: This looks insecure. Please use https://ruby-doc.org/stdlib-2.7.1/libdoc/tempfile/rdoc/Tempfile.html instead.

File: lib/puppet_litmus/puppet_helpers.rb [hunk 2]
```diff
@@ -147,6 +147,40 @@ def create_manifest_file(manifest)
...
+      tmp_file = File.open(tmp_filename, 'w')
+      tmp_file.write(content)
+      tmp_file.close
```
Comment (DavidS) [inline lib/puppet_litmus/puppet_helpers.rb:165]: It also looks like this is not removing the `tmp_file` after it is done. `tempfile` will address that too.

**Review by DavidS — CHANGES_REQUESTED** [review]
* [x] use tempfile
* [x] add at least basic unit tests to spec/lib/puppet_litmus/puppet_helpers_spec.rb

File: lib/puppet_litmus/puppet_helpers.rb (hunk 2 above)
Comment (RandomNoun7) [reply lib/puppet_litmus/puppet_helpers.rb:165]: With the way the existing `create_manifest_file` I thought that was the intended functionality. My temp directory is littered with them. I'll fix it in this new helper but I should probably go back and put up a maint to fix the existing helper.

Comment (DavidS) [reply lib/puppet_litmus/puppet_helpers.rb:165]: yes, please :heart:

File: lib/puppet_litmus/puppet_helpers.rb [hunk 3]
```diff
@@ -147,6 +147,48 @@ def create_manifest_file(manifest)
...
+          inventory_hash = inventory_hash_from_inventory_file
+          span.add_field('litmus.node_name', target_node_name)
+          add_platform_field(inventory_hash, target_node_name)
+
+          bolt_result = upload_file(tmp_file.path, destination, target_node_name, options: {}, config: nil, inventory: inventory_hash)
+          span.add_field('litmus.bolt_result.file_upload', bolt_result)
+          raise bolt_result.first['value'].to_s unless bolt_result.first['status'] == 'success'
+
+          if opts.key? :owner
+            bolt_result = run_shell("chown #{opts[:owner]} #{destination}")
```
Comment (michaeltlombardi) [inline lib/puppet_litmus/puppet_helpers.rb:181]: Is `owner` and opt you can pass to a Windows box?

Comment (RandomNoun7) [reply lib/puppet_litmus/puppet_helpers.rb:181]: Nope. This PR is an artifact of a solution I needed to make file writes on a master work. ACL's are so much more complex I figure Windows support for permissions will take it's own PR.

**Review by michaeltlombardi — APPROVED** [review]

**Pull request merged**